pkg/helm/updates.go

```diff
@@ -252,8 +271,40 @@ func GetReplicatedSecretFromUpstreamChartVersion(helmApp *apptypes.HelmApp, lice
 	return obj.(*corev1.Secret), nil
 }
 
-func pullChartVersion(helmApp *apptypes.HelmApp, licenseID string, version string) (*bytes.Buffer, error) {
-	err := CreateHelmRegistryCreds(licenseID, licenseID, helmApp.ChartPath)
+func GetReplicatedMetadataFromUpstreamChartVersion(helmApp *apptypes.HelmApp, licenseID string, version string) (*ReplicatedMeta, error) {
+	chartData, err := downloadChartReleaseIfNeeded(helmApp, licenseID, version)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get chart data")
+	}
+
+	valuesData, err := util.GetFileFromTGZArchive(chartData, "**/values.yaml")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get values.yaml from chart archive")
+	}
+
+	b := valuesData.Bytes()
+	values := struct {
+		Replicated *ReplicatedMeta `yaml:"replicated,omitempty"`
+	}{}
+	err = yaml.Unmarshal(b, &values)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to decode values.yaml")
+	}
+
+	return values.Replicated, nil
```

Comment:
Is it possible that `values.Replicated` is `nil` here?  If it's possible, it looks like it may panic when passing it to `helm.SetCachedUpdateMetadata(...)` [here](https://github.com/replicatedhq/kots/pull/3330/files#diff-5312bb96f485723d97db2688be857c2730d95bb339425202e587e69e4d8e4f66R467).

Reply:
yep, updated!